File: JavaToKotlin/app/src/main/java/io/dustin/apps/common/model/QueryPage.java
package io.dustin.apps.common.model;

import lombok.Data;
import lombok.ToString;
import lombok.extern.slf4j.Slf4j;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Sort;

@Data
@Slf4j
@ToString
public class QueryPage {

    private int page;
    private int size;
    private Long nextId;
    private String column;

    private Sort.Direction sort;

    public int offset() {
        return this.page - 1;
    }

    public int limit() {
        return this.size;
    }

    public PageRequest pageable() {
        Sort resultSort = Sort.unsorted();
        if(column != null && sort != null) {
            resultSort = Sort.by(this.sort, column);
        }
        return PageRequest.of(offset(), limit(), resultSort);
    }

}

File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/common/model/QueryPage.kt
package io.dustin.apps.common.model

import lombok.Data

@Data
@Slf4j
@ToString
class QueryPage {
    private val page = 0
    private val size = 0
    private val nextId: Long? = null
    private val column: String? = null
    private val sort: Sort.Direction? = null
    fun offset(): Int {
        return page - 1
    }

    fun limit(): Int {
        return size
    }

    fun pageable(): PageRequest? {
        var resultSort: Sort = Sort.unsorted()
        if (column != null && sort != null) {
            resultSort = Sort.by(sort, column)
        }
        return PageRequest.of(offset(), limit(), resultSort)
    }
}
